homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -133,54 +124,28 @@ def logout() -> None:
 
         def try_connect(user_input: dict[str, Any]) -> Connection:
             """Try connecting with given credentials."""
-            username = user_input.get(CONF_USERNAME)
-            password = user_input.get(CONF_PASSWORD)
-            conn: Connection
-            if username or password:
-                conn = AuthorizedConnection(
-                    user_input[CONF_URL],
-                    username=username,
-                    password=password,
-                    timeout=CONNECTION_TIMEOUT,
-                )
-            else:
-                try:
-                    conn = AuthorizedConnection(
-                        user_input[CONF_URL],
-                        username="",
-                        password="",
-                        timeout=CONNECTION_TIMEOUT,
-                    )
-                    user_input[CONF_USERNAME] = ""
-                    user_input[CONF_PASSWORD] = ""
-                except ResponseErrorException:
-                    _LOGGER.debug(
-                        "Could not login with empty credentials, proceeding unauthenticated",
-                        exc_info=True,
-                    )
-                    conn = Connection(user_input[CONF_URL], timeout=CONNECTION_TIMEOUT)
-                    del user_input[CONF_USERNAME]
-                    del user_input[CONF_PASSWORD]
+            username = user_input.get(CONF_USERNAME) or ""
+            password = user_input.get(CONF_PASSWORD) or ""
+            conn = AuthorizedConnection(
+                user_input[CONF_URL],
+                username=username,
+                password=password,
+                timeout=CONNECTION_TIMEOUT,
+            )
             return conn
 
-        def get_router_title(conn: Connection) -> str:
-            """Get title for router."""
-            title = None
+        def get_device_info(conn: Connection) -> dict[str, Any]:
+            """Get router info."""
             client = Client(conn)
             try:
-                info = client.device.basic_information()
+                return client.device.information()
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.debug("Could not get device.information", exc_info=True)
```

Comment:
If it's a truly unexpected exception, we should mark it as `error`. Catch all expected ones and don't error log those.

Reply:
This is kind of part of one of the things I meant with
>  There are a few things that can be cleaned up after this is merged, they're not included here to keep the PR smaller. Will submit them separately afterwards.

Now that we can assume an authorized connection here, we can clean up a bunch of things. It will inflate the size of the PR though, which is why I thought I'd leave it for later. But I can do that already here if you prefer it that way, I'd just rather not delay this more than necessary. Let me know if you want it in already now, otherwise I'll do it separately soon afterwards.